frontend/src/metabase/services.js

```diff
@@ -28,6 +32,35 @@ export const GTAPApi = {
   attributes: GET("/api/mt/user/attributes"),
 };
 
+export function maybeUsePivotEndpoint(api: APIMethod, card: Card): APIMethod {
+  function wrap(api) {
+    return (params: ?Data, ...rest: any) => {
+      const question = new Question(card, new Metadata());
+      const setting = question.setting("pivot_table.column_split");
+      const breakout = question.query().breakouts();
+      const { rows: pivot_rows, columns: pivot_cols } = _.mapObject(
+        setting,
+        fieldRefs =>
+          fieldRefs.map(field_ref =>
+            breakout.findIndex(b => _.isEqual(b, field_ref)),
+          ),
+      );
+      return api({ ...params, pivot_rows, pivot_cols }, ...rest);
+    };
+  }
+  if (card.display !== "pivot") {
+    return api;
+  }
+  switch (api) {
```

Comment:
Might be worth leaving a note as to why we have to check these.

Reply:
added!